src/pip/_internal/cli/cmdoptions.py

```diff
@@ -702,11 +702,7 @@ def _handle_build_dir(option, opt, value, parser):
     metavar='dir',
     action='callback',
     callback=_handle_build_dir,
-    help='Directory to unpack packages into and build in. Note that '
-         'an initial build still takes place in a temporary directory. '
-         'The location of temporary directories can be controlled by setting '
-         'the TMPDIR environment variable (TEMP on Windows) appropriately. '
-         'When passed, build directories are not cleaned in case of failures.'
+    help=SUPPRESS_HELP,
```

Comment:
I think it'd be better to add (deprecated) to the description, instead of suppressing it.

Reply:
My reasoning was to hide a deprecated option from new users and simplify the documentation, under the assumption that those who use it know what it is about.

Reply:
My concern with that if someone is researching how the replacement would work, supressing this option's help makes this information (about the current behavior of the flag) inaccessible for all users, not just new users.

I feel "you're using it therefore you know the nuances of how it works and how to replace it" is a very... not-friendly approach toward any user who's affected by this deprecation.

Reply:
Ok, done. I don't have a strong opinion on this. I personally find the help text for `--build` incomprehensible :cat:

Reply:
Thank you! ^>^

> I personally find the help text for --build incomprehensible 🐱

You're not alone. We'll all sigh collectively when we remove this option entirely. :)